pkg/virtualnode/membership_designator.go

```diff
@@ -33,8 +35,15 @@ type membershipDesignator struct {
 // +kubebuilder:rbac:groups=appmesh.k8s.aws,resources=virtualnodes,verbs=get;list;watch
 
 func (d *membershipDesignator) Designate(ctx context.Context, pod *corev1.Pod) (*appmesh.VirtualNode, error) {
+
+	// see https://github.com/kubernetes/kubernetes/issues/88282 and https://github.com/kubernetes/kubernetes/issues/76680
+	req := webhook.ContextGetAdmissionRequest(ctx)
+	podNS := &corev1.Namespace{}
+	if err := d.k8sClient.Get(ctx, types.NamespacedName{Name: req.Namespace}, podNS); err != nil {
```

Comment:
i don't think we need this namespace exists check here :D

Reply:
good catch! optimized